File: android/framework_old/src/main/java/com/dreampany/vision/ml/VisionImageProcessor.java
package com.dreampany.vision.ml;

import android.graphics.Bitmap;
import android.media.Image;

import com.google.firebase.ml.common.FirebaseMLException;

import java.nio.ByteBuffer;

/**
 * Created by Hawladar Roman on 9/28/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 */
public interface VisionImageProcessor {
    void process(ByteBuffer data, FrameMetadata meta, GraphicOverlay overlay) throws FirebaseMLException;

    void process(Bitmap bitmap, GraphicOverlay overlay);

    void process(Image image, int rotation, GraphicOverlay overlay);

    void stop();
}

File: android/word/src/main/java/com/dreampany/word/ml/vision/VisionImageProcessor.kt
package com.dreampany.word.ml.vision

import android.graphics.Bitmap
import android.os.Build.VERSION_CODES
import androidx.annotation.RequiresApi
import androidx.camera.core.ImageProxy
import com.dreampany.word.misc.graphic.GraphicOverlay
import com.google.mlkit.common.MlKitException
import java.nio.ByteBuffer

/**
 * Created by roman on 10/7/21
 * Copyright (c) 2021 epany. All rights reserved.
 * ifte.net@gmail.com
 * Last modified $file.lastModified
 */
interface VisionImageProcessor {
    fun processBitmap(bitmap: Bitmap, overlay: GraphicOverlay)

    @Throws(MlKitException::class)
    fun processByteBuffer(data: ByteBuffer, frameMeta: FrameMeta, overlay: GraphicOverlay)

    @RequiresApi(VERSION_CODES.KITKAT)
    @Throws(MlKitException::class)
    fun processImageProxy(image: ImageProxy, overlay: GraphicOverlay)

    fun stop()

}